src/IO/Exporter/PointwiseInterpolator.hpp

```diff
@@ -112,4 +114,83 @@ tuples::tagged_tuple_from_typelist<Tags> interpolate_to_points(
 }
 /// @}
 
+/*!
+ * \brief Interpolates data in volume files to target points by reading the
+ * volume data into memory
+ *
+ * This class reads the volume data at the requested time into memory and can
+ * then interpolate it to any number of target points.
+ */
+template <size_t Dim, typename Frame = ::Frame::Inertial>
+struct PointwiseInterpolator {
+  PointwiseInterpolator() = default;
+  PointwiseInterpolator(const std::variant<std::vector<std::string>,
+                                           std::string>& volume_files_or_glob,
+                        const std::string& subfile_name,
+                        const ObservationVariant& observation,
+                        const std::vector<std::string>& tensor_components);
+
+  /*!
+   * \brief Interpolate to many points
+   *
+   * \param result the interpolated data at the target points. The outer vector
+   * is the number of components and the inner vector is the number of target
+   * points. Will be resized automatically.
+   * \param target_points the points to interpolate to
+   * \param extrapolate_into_excisions whether to extrapolate into excised
+   * regions
+   * \param error_on_missing_points whether to throw an error if any of the
+   * target points are outside the domain
+   * \param num_threads the number of OpenMP threads to use to parallelize the
+   * interpolation over the target points. If not provided, the default number
+   * of threads will be used.
+   */
+  void interpolate_to_points(
+      gsl::not_null<std::vector<DataVector>*> result,
+      const tnsr::I<DataVector, Dim, Frame>& target_points,
+      bool extrapolate_into_excisions = false,
+      bool error_on_missing_points = false,
+      std::optional<size_t> num_threads = std::nullopt) const;
+
+  /*!
+   * \brief Interpolate to a single point
+   *
+   * This function is thread safe, so the interpolator can be constructed to
```

Comment:
What is this contrasting with?  Are the other function not thread safe?  That should be documented if so.